### PR TITLE
CLDR-16346 fix glitch in coverage charts

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-16346

common/properties/coverageLevels.txt
- 3 locales lowered in coverage, while one improved

tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java
- (techdebt) separated out from ShowLocaleCoverage, since the code is entirely orthogonal.
- In so doing, changed many public items to private

tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java
- (techdebt) separated out ChartLocaleGrowth, since the code is entirely orthogonal.
- In so doing, changed many public items to private
- Removed the contents of SUPPRESS_PATHS_AFTER_SUBMISSION. Those were in place for the late-breaking changes in v42.

tools/cldr-code/src/main/java/org/unicode/cldr/util/CoreCoverageInfo.java
- Three lines were producing the inverted results from previous changes to the file, so fixed those.

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+    private static CLDRConfig testInfo = ToolConfig.getToolInstance();
+    private static final SupplementalDataInfo SUPPLEMENTAL_DATA_INFO = testInfo.getSupplementalDataInfo();
+
+    private static org.unicode.cldr.util.Factory factory = testInfo.getCommonAndSeedAndMainAndAnnotationsFactory();
+    private static final CLDRFile ENGLISH = factory.make("en", true);
+
+    // added info using pattern in VettingViewer.
+
+    private static final RegexLookup<Boolean> HACK = RegexLookup.<Boolean> of(LookupType.STANDARD, RegexLookup.RegexFinderTransformPath)
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:60]: ideally `HACK` should get a new name or a description of how to remedy the hack.

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+            { VersionInfo.getInstance(2,0,1), 2011 },
+            { VersionInfo.getInstance(1,9,1), 2010 },
+            { VersionInfo.getInstance(1,7,2), 2009 },
+            { VersionInfo.getInstance(1,6,1), 2008 },
+            { VersionInfo.getInstance(1,5,1), 2007 },
+            { VersionInfo.getInstance(1,4,1), 2006 },
+            { VersionInfo.getInstance(1,3), 2005 },
+            { VersionInfo.getInstance(1,2), 2004 },
+            { VersionInfo.getInstance(1,1,1), 2003 },
+        };
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:173]: a third (fourth?) list of versions?

Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:173]: can you put a link to my unscheduled ticket https://unicode-org.atlassian.net/browse/CLDR-9149 (maybe we should schedlue it)

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+//            final Iterable<String> fullIterable = file.fullIterable();
+//            for (String path : fullIterable) {
+//                if (path.contains("\"one[@")) {
+//                    boolean inside = extra.contains(path);
+//                    Status status = new Status();
+//                    String loc = file.getSourceLocaleID(path, status );
+//                    int debug = 0;
+//                }
+//            }
+            // END HACK
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:294]:
```suggestion
```

dead code?

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:60]: needs description of what this is

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+public class ChartLocaleGrowth {
+
+    private static final boolean DEBUG = true;
+    private static final char DEBUG_FILTER = 0; // use letter to only load locales starting with that letter
+
+    private static CLDRConfig testInfo = ToolConfig.getToolInstance();
+    private static final SupplementalDataInfo SUPPLEMENTAL_DATA_INFO = testInfo.getSupplementalDataInfo();
+
+    private static org.unicode.cldr.util.Factory factory = testInfo.getCommonAndSeedAndMainAndAnnotationsFactory();
+    private static final CLDRFile ENGLISH = factory.make("en", true);
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:56]:
```suggestion
    private static final CLDRFile ENGLISH = testInfo.getEnglish();
```

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+            Counter<Level> missingCounter = new Counter<>();
+            Set<String> unconfirmedPaths = null;
+            Relation<MissingStatus, String> missingPaths = null;
+            unconfirmedPaths = new LinkedHashSet<>();
+            missingPaths = Relation.of(new LinkedHashMap<MissingStatus, Set<String>>(), LinkedHashSet.class);
+            VettingViewer.getStatus(latestFile.fullIterable(), file,
+                pathHeaderFactory, foundCounter, unconfirmedCounter,
+                missingCounter, missingPaths, unconfirmedPaths);
+
+            // HACK
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:306]: i think `HACK` means "don't count these against coverage" but more wording would be helpful here.

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+            }
+            for (Entry<MissingStatus, String> e : missingRemovals) {
+                missingPaths.remove(e.getKey(), e.getValue());
+            }
+            // END HACK
+
+            if (showMissing) {
+                int count = 0;
+                for (String s : unconfirmedPaths) {
+                    System.out.println(++count + "\t" + locale + "\tunconfirmed\t" + s);
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartLocaleGrowth.java:330]:
```suggestion
                    System.out.println(++count + "\t" + locale + "\t" + CLDRFile.DraftStatus.unconfirmed.name() +"\t" + s);
```

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java [hunk 7]
```diff
@@ -1194,91 +880,14 @@ private static class IterableFilter implements Iterable<String> {
...
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"prefix\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"given\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"given-informal\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"given2\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"surname-prefix\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"surname-core\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"surname2\"]",
-            "//ldml/personNames/sampleName[@item=\"foreignFull\"]/nameField[@type=\"suffix\"]"
+        private static final Set<String> SUPPRESS_PATHS_AFTER_SUBMISSION = ImmutableSet.of(
             );
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java:884]: should this be part of coverageLevels.xml?

**Review by srl295 — APPROVED** [review]
LGTM with suggestions

**Comment by macchiati** [comment]
I'll follow up with a PR regarding the comments you made.

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java (hunk 7 above)
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java:884]: It's too painful to do in coverageLevels.

**Pull request merged**